### PR TITLE
[JS][BiDi] Add Script module commands and types

### PR DESCRIPTION
**Pull request by TamsilAmani**
### Description

Add script module commands and types

### Motivation and Context

Implement script module commands and types as described in https://w3c.github.io/webdriver-bidi/#module-script.
The majority of classes and interfaces are mapping the common data types as described in https://w3c.github.io/webdriver-bidi/#data-types.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist

- [x] I have read the [contributing](https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by TamsilAmani** [comment]
I suggest to add getter functions where ever possible instead of directly accessing the object's attributes. Is that okay?

File: javascript/node/selenium-webdriver/bidi/evaluateResult.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+// under the License.
+
+const EvaluateResultType = {
+  SUCCESS: 'success',
+  EXCEPTION: 'exception',
+}
+
+class EvaluateResultSuccess {
+  constructor(type, realmId, value) {
+    this.resultType = type
```
Comment (harsha509) [inline javascript/node/selenium-webdriver/bidi/evaluateResult.js]: if this class only is for success , can we do `this.resultType = EvaluateResultType.SUCCESS;` right ?

File: javascript/node/selenium-webdriver/bidi/evaluateResult.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    this.realmId = realmId
+    this.result = value
+  }
+}
+
+class EvaluateResultException {
+  constructor(type, realmId, exceptionDetails) {
+    this.resultType = type
```
Comment (harsha509) [inline javascript/node/selenium-webdriver/bidi/evaluateResult.js]: same here `this.resultType = EvaluateResultType.EXCEPTION;` we can reduce one param on usage

File: javascript/node/selenium-webdriver/bidi/protocolType.js [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+const PrimitiveType = {
+  UNDEFINED: 'undefined',
+  NULL: 'null',
+  STRING: 'string',
+  NUMBER: 'number',
+  SPECIAL_NUMBER: 'number',
+  BOOLEAN: 'boolean',
+  BIGINT: 'bigint',
+
+  findByName: function (name) {
```
Comment (harsha509) [inline javascript/node/selenium-webdriver/bidi/protocolType.js]: another way (only if we want to reduce code)

  `findByName(name) {
    return Object.values(this).find((type) => {
      return typeof type === 'string' && name.toLowerCase() === type.toLowerCase();
    }) || null;
  },`

File: javascript/node/selenium-webdriver/bidi/protocolValue.js [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+      this.sharedId = remoteValue['sharedId']
+    }
+
+    if (this.value != null) {
+      this.value = this.deserializeValue(this.value, this.type)
+    }
+  }
+
+  deserializeValue(value, type) {
+    if (type === NonPrimitiveType.MAP || type === NonPrimitiveType.OBJECT) {
```
Comment (harsha509) [inline javascript/node/selenium-webdriver/bidi/protocolValue.js]: can we use like below

  `if ([NonPrimitiveType.MAP, NonPrimitiveType.OBJECT].includes(type)) {
    return Object.fromEntries(value);
  } 
  else if ()...`

File: javascript/node/selenium-webdriver/bidi/protocolValue.js [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+    if (handle === RemoteReferenceType.HANDLE) {
+      this.handle = shareId
+    } else {
+      this.handle = handle
+      this.shareId = shareId
+    }
+  }
+
+  asMap() {
+    let toReturn = {}
```
Comment (harsha509) [inline javascript/node/selenium-webdriver/bidi/protocolValue.js]: const

**Comment by TamsilAmani** [comment]
Thank you for the review @harsha509. I have pushed the changes.

**Review by harsha509 — APPROVED** [review]
LGTM!

**Pull request merged**